_posts/2019-12-26-policy-gradients.md

```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+title:  "Understanding Policy Gradients"
+date:   2019-12-26 22:19:00 +0530
+categories: reinforcement-learning, policy-gradients
+---
+
+# Understanding Policy Gradients
+In reinforcement learning, we have an agent interacting with the environment. The agent acts, the environment gives feedback in the for of rewards. The agent wants to maximize the total rewards it gets.
+
+The goal goal of policy gradients: make an agent (a Neural Network, to be concrete) that maximizes its expected return. To be precise, the agent must: for every state it might be in, assign a probability distribution for actions it might take, so that it gets the maximum possible return on average from then onwards.
+
+## Vanilla Policy Gradient (REINFORCE)
+If we want the agent to get the maximum return on average, a natural first step is to directly optimize the policy(a NN that takes in state and outputs action probs).
+
+Find Ө to maximize Utility
+
+$$\mathop{maximize}_{\theta}\ \mathop{\mathbb{E}}_{\tau \char`\~ \pi_{\theta}}[R(\tau)])$$
+
+$$(Utility\ U(\pi_{\theta}) =\mathop{\mathbb{E}}_{\tau \char`\~  \pi_{\theta}}[R(\tau)]) $$
+
+Let $$\inline \pi_{\theta}$$ be our policy network. We can do gradient _ascent_ on $$\inline U(\pi)$$, wrt the policy net’s parameters, taking small steps in a direction which is likely to increase the return
+
+$$\theta_{k+1} \leftarrow \theta_{k} + \eta \nabla_{\theta} U(\pi_{\theta})$$
+
+We need the gradient term to be in a more useful form:
+
+$$\nabla_{\theta} U(\pi_{\theta})\ = \nabla_{\theta}\mathop{\mathbb{E}}_{\tau \char`\~ \pi_{\theta}}[R(\tau)])$$
+
```

Comment:
Explain what each term means in the equation.

Reply:
Added more explanations